test/jeison-test.el

```diff
@@ -234,6 +234,32 @@
                                   }"
                           '(a (jeison:filter-candidates b) number))))))
 
+(ert-deftest jeison:check-hash-table-of-invalid-key-type ()
+  ;; NOTE: EIEIO checks the type constraint when it creates a new instance,
+  ;; not when the code invokes 'defclass'.  Consequently, EIEIO emits the
+  ;; 'invalid-slot-type' signal __only__ when the code invokes
+  ;; 'make-instance'.
+  (should (and (condition-case ()
+                   (progn
+                     (jeison-defclass jeison:treif-key-type-class ()
+                       ((foo :initarg :foo :type (jeison-hash-table-of integer string))))
+                     (make-instance jeison:treif-key-type-class :foo (make-hash-table))
```

Comment:
I see why you hesitated, it is indeed weird to check `make-instance` since it's not part of Jeison.
Jeison checks (or should check) for expected types in `jeison-read` and that's what should be tested.

Reply:
`make-instance` is the *only* way to trigger the `invalid-slot-type` signal and EIEIO type checking, which is what I took away from your comments. Now do you understand why I asked you to be precise?

You want (a) code in `jeison---read-internal` to validate the key type is either `string` or `symbol`, (b)  a test to exorcise that code (yes, the pun is intended.)

Reply:
It is again a discussion similar to hash table keys. No need to go down to details, look at it from the interface perspective. The user has expected types and some JSON. Jeison maps JSON to user expectations. If the expectation is not correct, it is Jeison's job to surface that.

Let's look at this code here: https://github.com/SavchenkoValeriy/jeison/blob/19a51770f24eaa7b538c7be6a8a5c25d154b641f/jeison.el#L170-L173
That is one example of where we do it.

It is possible that the user expectations expressed in the form of static types do not match the actual JSON. Jeison does have tests for this, here's an example:
https://github.com/SavchenkoValeriy/jeison/blob/19a51770f24eaa7b538c7be6a8a5c25d154b641f/test/jeison-test.el#L114-L120

With new types, user expectations can also be wrong. It is APIs job to surface any problem we can spot. Let me repost that snippet from before:
```emacs-lisp
(jeison-defclass jeison:jeison-class nil ((x :initarg :x :type (jeison-hash-table-of string string)))
(jeison-read jeison:jeison-class "{\"x\": 42}")
```
`42` cannot be parsed as a hash table at all, we need to report it.
Let's go further:
```js
{"x": [0, 1], "y": [2], "z": [3, 4, 5]}
```
It looks like a hash table, now! But if the user asked for `jeison-hash-table-of string string`, we have a problem and we need to report it.
And yet another example:
```js
{"x": "a", "y": [2], "z": "b"}
```
that we can use to test type mismatch for one of the values.

When the user writes `jeison-hash-table-of NOT_STRING_OR_SYMBOL T`, it is another example of wrong expectations. Yes, it is different from other cases. Yes, we can check it not during the reading part, but during the class declaration. But it's all implementation details. The main part is that it's wrong, it should be checked, it should be reported, it should be tested.

I want tests for that, and not for `make-instance` or `gethash`.